packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1266,8 +1266,14 @@ class _TabBarViewState extends State<TabBarView> {
       return Future<void>.value();
 
     final int previousIndex = _controller.previousIndex;
-    if ((_currentIndex - previousIndex).abs() == 1)
-      return _pageController.animateToPage(_currentIndex, duration: kTabScrollDuration, curve: Curves.ease);
+    if ((_currentIndex - previousIndex).abs() == 1) {
+      _warpUnderwayCount += 1;
+      await _pageController.animateToPage(_currentIndex, duration: kTabScrollDuration, curve: Curves.ease);
+      if (!mounted)
```

Comment:
This is not needed, we do this check for warpping > 1 tab because it is physically changing the children order.

Reply:
Aha! I get it. Will remove this check, thanks!